### PR TITLE
Measure SDK size overhead

### LINKED ISSUE #2437: Document SDK size overhead
### Description

Measure size overhead and document it

```[tasklist]
### Tasks
- [x] Readme
- [ ] Sentry docs in https://docs.sentry.io/platforms/flutter/overhead/
```

### PR DESCRIPTION
**Pull request by denrase**
## :bulb: Motivation and Context

Closes #2437

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: .github/workflows/benchmarking.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+jobs:
+  android:
+    name: Collect Android metrics
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: subosito/flutter-action@f2c4f6686ca8e8d6e6d0f28410eeef506ed66aff # pin@v2.18.0
+
+      - name: build sample-flutter
+        working-directory: ./benchmarking/sample_flutter
```
Comment (vaind) [inline .github/workflows/benchmarking.yml:25]: Can we avoid adding another full-fledged sample app to the repo and instead use `flutter create`? That also saves us from updating every now & then.

Comment (denrase) [reply .github/workflows/benchmarking.yml:25]: good point, thank you, just need to make sure everything is patched correctly to create the ios ipa, as we need to sign it afaik

Comment (vaind) [reply .github/workflows/benchmarking.yml:25]: you can pass `--no-codesign` (or some such) which would make it easier. Although it's fair to check first if it affects the size but that can be done locally on any app

Comment (denrase) [reply .github/workflows/benchmarking.yml:25]: @vaind I just saw the [Metrics PR](https://github.com/getsentry/sentry-dart/pull/1034) that you have created, which makes this here pretty much redundant, should have looked before. 🤦

**Comment by denrase** [comment]
Closing, as this is a duplicate of #1034

**Pull request closed without merging**